functional-tests/fn_el_schnorr_precompile.py

```diff
@@ -0,0 +1,75 @@
+import hashlib
+
+import flexitest
+from strata_utils import sign_schnorr_sig
+from web3 import Web3
+
+import testenv
+from constants import PRECOMPILE_SCHNORR_ADDRESS
+from utils import wait_until_with_value
+
+
+@flexitest.register
+class SchnorrPrecompileTest(testenv.StrataTester):
+    def __init__(self, ctx: flexitest.InitContext):
+        ctx.set_env("basic")
+
+    def main(self, ctx: flexitest.RunContext):
+        reth = ctx.get_service("reth")
+        self.web3: Web3 = reth.create_web3()
+
+        self.source = self.web3.address
+        self.dest = self.web3.to_checksum_address(PRECOMPILE_SCHNORR_ADDRESS)
+
+        # secret key
+        secret_key = "a9f913c3d7fe56c462228ad22bb7631742a121a6a138d57c1fc4a351314948fa"
+        self.debug(secret_key)
+
+        message_hash = hashlib.sha256(b"AlpenStrata").hexdigest()
+        (signature, public_key) = sign_schnorr_sig(message_hash, secret_key)
+
+        valid_precompile_input = public_key + message_hash + signature
+        data = self.call_precompile_and_get_return_data(valid_precompile_input)
+        assert data == "0x01", f"Schnorr verification failed: expected '0x01', got '{data}'."
+
+        invalid_message = hashlib.sha256(b"MakaluStrata").hexdigest()
+        invalid_precompile_input = public_key + invalid_message + signature
+        data = self.call_precompile_and_get_return_data(invalid_precompile_input)
+        assert data == "0x00", f"Schnorr verification failed: expected '0x00', got '{data}'."
+
+        return True
+
+    def call_precompile_and_get_return_data(self, precompile_input: str):
```

Comment:
Maybe something like this?
```suggestion
    def wait_for_precompile_response(self, precompile_input: str):
```
A function implicitly returns data so the name doesn't need to have "return".